Weird Python highlighting

### Summary

The following Python code has very weird highlights:
```python
K = 0
X = 0
XY = 0
k = 0
XYZ = 0
ZF00D = 0
zf00d = 0
BAR = 0
KV = 0
A0 = 0
_A0 = 0
```

For some reason, helix seems to have its own set of custom highlights in `runtime/queries/python/highlights.scm` (I'm assuming `((identifier) @constant (#match? @constant "^[A-Z_]{2,}$"))` is at least partly responsible for this.

Why does helix need such a file though? Shouldn't tree-sitter be enough?

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. `hx /tmp/x.py`
2. enter the above snippet

I expected this to happen: same color for all identifiers (at least the same for all capitalized ones)

Instead, this happened:
![2023-04-12-232132-Fool4ugi](https://user-images.githubusercontent.com/34467/231587903-a2a72fff-dc06-486b-8ae4-26dc24a13c29.png)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-04-12T23:25:56.536 helix_vcs [ERROR] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepositoryWithinFs {
            path: "/tmp",
            limit: "/",
        },
    ),
}
2023-04-12T23:25:56.536 helix_vcs [ERROR] failed to open diff base for /tmp/x.py
2023-04-12T23:25:56.537 helix_vcs [ERROR] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepositoryWithinFs {
            path: "/tmp",
            limit: "/",
        },
    ),
}
2023-04-12T23:25:56.537 helix_vcs [ERROR] failed to obtain current head name for /tmp/x.py
2023-04-12T23:25:56.676 helix_term::application [ERROR] Language Server: Received malformed method call workspace/configuration in request 0: protocol error: InvalidParams: Invalid params: invalid value: string "", expected relative URL without a base.
```

</details>


### Platform

Linux

### Terminal Emulator

alacritty 0.12.0 (5a728195)

### Helix Version

helix 23.03

--- a/runtime/queries/python/highlights.scm
+++ b/runtime/queries/python/highlights.scm
@@ -94,7 +94,7 @@
 ; Variables
 
 ((identifier) @constant
- (#match? @constant "^[A-Z_]{2,}$"))
+ (#match? @constant "^_*[A-Z][A-Z\\d_]*$"))
 
 ((identifier) @type
  (#match? @type "^[A-Z]"))

